addons/binding/org.openhab.binding.feican/src/main/java/org/openhab/binding/feican/internal/FeicanDiscoveryService.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.feican.internal;
+
+import static org.openhab.binding.feican.FeicanBindingConstants.*;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for Feican Bulbs. When sending a discovery UDP broadcast command on port 5000 to a Feican bulb. The
+ * bulp will respond with it's mac address send via UDP broadcast over port 6000.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, immediate = true)
+public class FeicanDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int DISCOVERY_TIMEOUT_SECONDS = 5;
+    private static final int RECEIVE_JOB_TIMEOUT = 20000;
+    private static final int UDP_PACKET_TIMEOUT = RECEIVE_JOB_TIMEOUT - 50;
+    private static final String FEICAN_NAME_PREFIX = "FC_";
+
+    private final Logger logger = LoggerFactory.getLogger(FeicanDiscoveryService.class);
+
+    ///// Network
+    private final byte[] buffer = new byte[32];
+    private final DatagramPacket receivePacket;
+    private final DatagramPacket discoverPacket;
+    @Nullable
+    private DatagramSocket discoverSocket;
+
+    public FeicanDiscoveryService() throws UnknownHostException {
```

Comment:
> throws UnknownHostException

Err, what? That must not be done by a constructor of a DiscoveryService.

Reply:
This is caused by the call `InetAddress.getByName("255.255.255.255");` in the constructor. I assume the throws is not desired, but how should I handle the exception?

Reply:
Probably best to log a warning and then simply don't be functional, but not throwing an exception, which can result in ugly log messages in DS.